### ISSUE TITLE
Test_TC_LVL_2_1 influenced by a prior Darwin Framework Tests

### ISSUE THREAD

**Issue by jmartinez-silabs** [project-chip/connectedhomeip#13551]
#### Problem
PR #13338 enables optional features for the level control and on-off cluster. Those two clusters become inter-connected in some cases with on/off level transition.

What I debugged so far is that some prior test causes Level Control on level attribute to change from 254 to 1. This will then cause the On (or Toggle to On) command to also set the current level to 1. The On command is used in a few tests for example the color control clusters tests.

I could not find yet which test modifies on level attribute but before any test cases are run, the attribute value is correct. Also, this problematic scenario only occurs in Darwin Framework Tests and not the Linux test suites. Are there additional tests run in Darwin framework tests?

To unblock #13338, I made a temporary band-aid, I reset the CurrentLevel to the current default value at the beginning of TC_LVL_2_1.

I create this issue to track this band-aid as I continue to debug for in a separate PR.

**Comment by bzbarsky-apple** [comment]
The framework tests that ran multiple YAMLs without a stack restart are gone.
